src/qmapshack/dem/CDemVRT.cpp

```diff
@@ -412,6 +412,57 @@ void CDemVRT::draw(IDrawContext::buffer_t& buf)
                     drawTile(img, r, p);
                     p.setOpacity(o1);
                 }
+
+                if(doElevationShading())
+                {
+                    QPolygonF r = l;
+
+                    QImage img(w_used, h_used, QImage::Format_Indexed8);
+                    img.setColorTable(elevationShadeTable);
+
+                    elevationshading(data, w_used, h_used, img);
+
+                    drawTile(img, r, p);
+                }
+
+            }
+        }
+
+        // draw the elevation shade scale
+        if(doElevationShading())
```

Comment:
make it a method please. This is already too much spaghetti code without these additional lines.